src/middlewared/middlewared/api/v25_10_0/pool_dataset.py

```diff
@@ -581,3 +581,17 @@ class PoolDatasetUpdateArgs(BaseModel):
 
 class PoolDatasetUpdateResult(BaseModel):
     result: PoolDatasetEntry
+
+
+class PoolDatasetRenameOptions(BaseModel):
+    new_name: NonEmptyString
```

Comment:
To match schema of `pool.dataset.create`:
```suggestion
    new_name: NonEmptyString = Field(max_length=ZFS_MAX_DATASET_NAME_LEN)
```